### ISSUE TITLE
Feature request: enforce file name conventions

### ISSUE THREAD

**Issue by danvk** [jest-community/eslint-plugin-jest#751]
I realized recently that my project uses several different conventions for what files containing Jest tests are named:

1. `dir/__tests__/module.test.ts` (plural tests)
1. `dir/__test__/module.test.ts` (singular test)
1. `dir/module.test.ts`

I'd like to enforce that we only use the first form (`__tests__`). This seems like the sort of thing that eslint-plugin-jest could help with. Are test file names in scope for the project? Is there a suggested alternative?

**Comment by k-yle** [comment]
You can already do this using overrides in your eslint config to only set `env: { jest: true }` in the files you want. You don't even need this plugin:

```js
{
  overrides: [
    {
      files: ["**/__tests__/*.test.*"],
      env: {
        jest: true,
      },
      plugins: ["jest"],
      extends: ["plugin:jest/recommended"]
    }
  ]
}
```

then eslint will error if you use jest globals (`expect`, `describe`, `it` etc.) in any other files

**Comment by G-Rath** [comment]
Actually, I think the way you should do this would be to set [`testRegex`](https://jestjs.io/docs/en/configuration#testregex-string--arraystring) to match only the test files you want.

You could back it up with an `overrides` as @k-yle says to have ESLint tell you if you're using a `jest` global outside of those files.

**Comment by G-Rath** [comment]
I'm going to close this as I think the above are the best ways to do this + it's a bit of a chicken and egg situation (we could in theory report, but that would require being applied to non-test files when our rules expect to only be running on test files...).

You should be able to back this up with a plugin like [`eslint-plugin-filename-rules`](https://www.npmjs.com/package/eslint-plugin-filename-rules).
